### PR TITLE
Added some test cases to deepForEach

### PR DESCRIPTION
**Pull request by deflucaseng**
A comprehensive set of test cases for the deepForEach function located in src/utils/deepForEach.js. The new tests ensure that the function correctly iterates over elements in various types of arrays and matrix objects, validating its behavior across different scenarios.

Test Cases Added:
Simple Array: Validates iteration over a flat array of numbers.
Nested Array: Ensures correct iteration through multiple levels of nested arrays.
Mixed Type Array: Tests iteration over an array containing different types of elements, including numbers, strings, null, undefined, and booleans.
Empty Array: Confirms the function handles an empty array without errors.
Array with Empty Nested Arrays: Checks iteration through arrays that contain empty nested arrays.
DenseMatrix: Verifies iteration over elements within a DenseMatrix object.
Nested DenseMatrix: Tests iteration over nested DenseMatrix objects.
Matrix Conversion: Ensures the function works correctly with Matrix objects by converting them to arrays and iterating through their elements.
Complex Numbers: Confirms correct iteration over arrays containing complex number elements.

### CODE REVIEW

**Comment by josdejong** [comment]
Thanks Lucas, this looks very well tested!

I see there are two minor linting issues (you can run `npm run lint` to see). Can you please run `npm run format` to fix them and commit the changes?

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "core-js": "3.37.0",
     "del": "6.1.1",
     "dtslint": "4.2.1",
-    "eslint": "8.57.0",
+    "eslint": "^8.57.0",
```
Comment (josdejong) [inline package.json:56]: Can you please revert the changes in `package.json` and `package-lock.json`?

Comment (deflucaseng) [reply package.json:56]: My apologies I am pretty new to lint

Comment (josdejong) [reply package.json]: 👍 np

**Comment by josdejong** [comment]
All looks good, thanks again 👌
